src/vm/database/clarity_db.rs

```diff
@@ -561,10 +584,7 @@ impl<'a> ClarityDatabase<'a> {
     pub fn get_current_burnchain_block_height(&mut self) -> u32 {
         let cur_stacks_height = self.store.get_current_block_height();
         let last_mined_bhh = if cur_stacks_height == 0 {
-            StacksBlockHeader::make_index_block_hash(
-                &FIRST_BURNCHAIN_CONSENSUS_HASH,
-                &FIRST_STACKS_BLOCK_HASH,
-            )
+            return self.burn_state_db.get_burn_start_height();
```

Comment:
While this should be correct (I checked in prod), this feels risky.  Do we at least want to assert somewhere that `.get_burn_start_height()` returns the same value as `.get_burnchain_block_height()` on the initial block hash?

Reply:
In the genesis block, `.get_burnchain_block_height()` would previously _panic_, which wasn't previously a problem, because none of the contracts deployed or called in the genesis block queried the burnchain block height. However, because the versioning check _does_ query the burnchain block height during the genesis initialization, that case had to be handled.